cognite_toolkit/_cdf_tk/storageio/_applications.py

```diff
@@ -30,29 +30,15 @@ def stream_data(self, selector: ChartSelector, limit: int | None = None) -> Iter
             ...
         elif isinstance(selector, ChartOwnerSelector):
             selected_charts = ChartList([chart for chart in selected_charts if chart.owner_id == selector.owner_id])
+        elif isinstance(selector, ChartExternalIdSelector):
+            external_id_set = set(selector.external_ids)
+            selected_charts = ChartList([chart for chart in selected_charts if chart.external_id in external_id_set])
         else:
             raise ToolkitNotImplementedError(f"Unsupported selector type {type(selector).__name__!r} for ChartIO")
 
         if limit is not None:
             selected_charts = ChartList(selected_charts[:limit])
         for chunk in chunker_sequence(selected_charts, self.CHUNK_SIZE):
-            ts_ids_to_lookup = {
-                ts_ref.ts_id
-                for chart in chunk
-                for ts_ref in chart.data.time_series_collection or []
-                if ts_ref.ts_external_id is None and ts_ref.ts_id is not None
-            }
-
-            if ts_ids_to_lookup:
-                retrieved_ts = self.client.time_series.retrieve_multiple(
-                    ids=list(ts_ids_to_lookup), ignore_unknown_ids=True
-                )
-                id_to_external_id = {ts.id: ts.external_id for ts in retrieved_ts}
-
-                for chart in chunk:
-                    for ts_ref in chart.data.time_series_collection or []:
-                        if ts_ref.ts_id in id_to_external_id:
-                            ts_ref.ts_external_id = id_to_external_id[ts_ref.ts_id]
```

Comment:
Moved this to data_to_json_chunk as the stream should have unchanged charts as output.